src/main/java/Duke.java

```diff
@@ -1,10 +1,208 @@
+import Task.Task;
+import Task.Todo;
+import Task.Deadline;
+import Task.Event;
+import Exception.InvalidCommandException;
+
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
 public class Duke {
+
+    static String filePath = "tasks.txt";
+    static Scanner in = new Scanner(System.in);
+    static ArrayList<Task> tasks = new ArrayList<>();
+
+    private static void writeToFile(String filePath, String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    public static void saveToFile() {
+        String fullTextToAdd = "";
+        try {
+            for (Task task: tasks) {
+                if (task != null) {
+                    fullTextToAdd += task.toRawData();
+                    fullTextToAdd += System.lineSeparator();
+                }
+            }
+            writeToFile(filePath, fullTextToAdd);
+        } catch (IOException e) {
```

Comment:
Try to make the happy path prominent

Reply:
What do you mean by prominent? The only edge case is if task==null and I shouldn't flip this if statement right?

Reply:
yes. handle the edge case first. then automatically, happy path becomes the main thing to see in the method.